Lib/test/test_string.py

```diff
@@ -475,6 +475,55 @@ class PieDelims(Template):
         self.assertEqual(s.substitute(dict(who='tim', what='ham')),
                          'tim likes to eat a bag of ham worth $100')
 
+    def test_is_valid(self):
+        eq = self.assertEqual
+        s = Template('$who likes to eat a bag of ${what} worth $$100')
+        valid = s.is_valid()
+        eq(valid, True)
```

Comment:
What about `self.assertTrue(s.is_valid())` here and `self.assertFalse(s.is_valid()))` below?